src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -117,7 +113,7 @@ public async Task Stop()
             consumer.Received -= Consumer_Received;
             messageProcessing.Cancel();
 
-            while (semaphore.CurrentCount != maxConcurrency)
+            while (Interlocked.Read(ref numberOfExecutingReceives) > 0)
```

Comment:
This condition seems fine. It shouldn't be possible for it to be less than zero, but if it was somehow we'd still want to skip the loop.